### PR TITLE
fix(conversations): emit conversation.opened when legacy auto assignment runs on the open transition

### PR DESCRIPTION
**Pull request by tds-1**
When a conversation transitioned to Open on an inbox using legacy (V1) auto-assignment, the assignment write ran as a second save on the same conversation inside the same transaction. That second save replaced `saved_changes`, so the commit callbacks only saw the assignee change: `conversation.opened` and `conversation.status_changed` were never emitted — no "Conversation opened" activity message, no `conversation_opened` reporting event, and automations triggered on Conversation Opened silently never ran. Since this happened exactly when auto-assignment succeeded, it was the dominant path on V1 inboxes, not an edge case (in a concurrency test, 58 of 60 open transitions lost the event). The most visible instance: an agent replying to a Captain Pending conversation, which opens it — but the same loss happened from the UI open button or any other open transition.

## Closes

- [CW-7967](https://linear.app/chatwoot/issue/CW-7967/conversation-opened-event-not-emitted)

## How to reproduce

1. On an inbox with auto-assignment enabled and Assignment V2 off, have at least one available agent.
2. Take an unassigned Pending conversation (e.g. one Captain is handling) and send a public agent reply, or open it via the status toggle.
3. The agent gets assigned, but the conversation shows no "Conversation opened" activity, reporting records no `conversation_opened` event, and Conversation Opened automations don't fire.

## What changed

- Legacy (V1) auto-assignment for status transitions now runs in a `before_save` and sets the assignee on the in-flight save, so status and assignee commit as a single change-set and both stay visible to the `after_commit` callbacks. Rails only surfaces one change-set per row per transaction to commit callbacks, so a follow-up save can never preserve both — folding the assignment into the same UPDATE is the shape that fixes this without touching the event layer.
- The lock-then-recheck semantics from #15275 are preserved and now genuinely serialize: `AgentAssignmentService#assign_under_lock` takes the row lock, re-checks `reassignment_still_needed?` against committed state, then sets the assignee — with the lock now acquired before our own write (previously the status UPDATE had already locked the row before the service's lock ran). Verified under concurrent load: opens racing manual assignment writes produce exactly one assignment write, no assignee bouncing, and no deadlocks, matching current behavior.
- New conversations keep the post-save assignment path (`perform`, now built on `assign_under_lock`): their status is only finalized by `before_create` callbacks, which run after `before_save`, and creates don't emit `conversation.opened` in the first place.
- Adds a regression spec asserting the open transition with V1 auto-assignment dispatches both `conversation.opened` and `assignee.changed`, with `status` and `assignee_id` in the same set of committed changes.

### CODE REVIEW

File: app/services/auto_assignment/agent_assignment_service.rb [hunk 1]
```diff
@@ -8,19 +8,27 @@ def find_assignee
     round_robin_manage_service.available_agent(allowed_agent_ids: allowed_online_agent_ids)
   end
 
+  # Locks the row to serialize with concurrent assignment writers, then sets the assignee
+  # on the in-memory conversation without saving. Called from the conversation's own
+  # before_save so status and assignee commit as one change-set and both stay visible to
+  # the after_commit callbacks. Returns the new assignee, or nil when nothing changed.
+  def assign_under_lock
+    locked = Conversation.lock.find_by(id: conversation.id)
+    return unless locked && reassignment_still_needed?(locked)
```
[the remark this answers is not in the document]
Comment (tds-1) [reply app/services/auto_assignment/agent_assignment_service.rb]: Fixed in 827c4fb6ce

**Review by sony-mathew — APPROVED** [review]
looks good overall, please take a look at this finding:

- **P2 — New, blocking: concurrent stale open transitions emit duplicate events.** In [`assign_under_lock`](/worktrees/5b68/chatwoot/app/services/auto_assignment/agent_assignment_service.rb:15), the second opener waits for the first, sees its valid assignee, and returns—but its stale `pending → open` change remains dirty. Its outer save therefore dispatches another `conversation.opened` and `conversation.status_changed`.

  Reproduction with two concurrent requests that loaded the same pending conversation:

  - Base `89a933f7`: 1 opened, 1 status-changed, 1 assignee event.
  - Head `827c4fb6`: **2 opened, 2 status-changed**, 1 assignee event.

  This can execute Conversation Opened automations twice and insert duplicate reporting rows because neither [`AutomationRuleListener`](/worktrees/5b68/chatwoot/app/listeners/automation_rule_listener.rb:39) nor [`ReportingEventListener`](/worktrees/5b68/chatwoot/app/listeners/reporting_event_listener.rb:128) deduplicates these events.
